### PR TITLE
RMB-591: Explain totalRecords estimation

### PR DESCRIPTION
**Pull request by julianladisch**
Also fix a border case: est_count < ${exactCount}

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1763,6 +1761,40 @@ A CQL querying example:
...
+
+RMB adds a `totalRecords` field to result sets. It contains an estimation how many records were matching if paging parameters were set to `offset` = 0 and `limit` = unlimited.
+
+It uses this algorithm:
+
+1. Run "EXPLAIN SELECT" to get an estimation from PostgreSQL.
+2. If this is greater than 4*1000 return it and stop.
+3. Run "SELECT COUNT(*) FROM query LIMIT 1000".
+4. If this is less than 1000 return it (this is the exact number) and stop.
+5. If the result from 1. is less than 1000 return 1000 and stop.
```
Comment (craigmcnally) [inline README.md:1774]: might be worth explicitly noting (this is where estimate < 1000 and actual >= 1000)

File: README.md [hunk 2]
```diff
@@ -1763,6 +1761,40 @@ A CQL querying example:
...
+
+1000 is configurable, see the [Post Tenant API](#the-post-tenant-api) how to set `exactCount` in schema.json.
+
+RMB adjusts `totalRecords` when the number of returned records in the current result set and the paging parameters `offset` and `limit` proves it wrong:
+
+* If no record is returned and `totalRecords > offset` then adjust `totalRecords = offset`.
+* If the number of records returned equals `limit` and `totalRecords < offset + limit` then adjust `totalRecords = offset + limit`.
+* If the number of records is at least one but less than `limit` then adjust `totalRecords = offset +` number of records returned (this is the exact number).
+
+Note that clients should continue on the next page when `totalRecords = offset + limit` because there may be more records.
```
Comment (craigmcnally) [inline README.md:1788]: This is important - might be worth highlighting (bold?)

File: README.md (hunk 1 above)
Comment (julianladisch) [reply README.md:1774]: Added totalRecords=1000 explanation.

File: README.md (hunk 2 above)
Comment (julianladisch) [reply README.md]: Bolded.

**Pull request closed without merging**